src/main/java/com/google/devtools/build/lib/bazel/repository/cache/RepositoryCache.java

```diff
@@ -298,7 +299,8 @@ public static void assertFileChecksum(XattrProvider xattrProvider, String expect
 
     String actualChecksum;
     try {
-      if (xattrProvider != null) {
+      if (xattrProvider != null
+          && filePath.getFileSystem().getDigestFunction().getNames().contains(keyType.toString())) {
```

Comment:
I'm not happy with `toString()` here but that's what currently works since `KeyType` and `DigestHashFunction` disagree on name casing:
 https://github.com/sushain97/bazel/blob/c3e2d296f13f6c0c044a68035e912a2173656fa3/src/main/java/com/google/devtools/build/lib/bazel/repository/cache/RepositoryCache.java#L44-L47

https://github.com/sushain97/bazel/blob/c3e2d296f13f6c0c044a68035e912a2173656fa3/src/main/java/com/google/devtools/build/lib/vfs/DigestHashFunction.java#L68-L69

I could add the lowercase versions (e.g. `sha256`) to `altNames` if you feel that's a better approach.

Reply:
you could just expose the underlying `HashFunction` and compare if they're equal.

Reply:
as you said, they're singletons in effect. Also, even if they weren't, false negatives are okay -- we'd just fall back to actually computing the digest again.

Reply:
OK, sounds good!